coordinator/history/history.go

```diff
@@ -0,0 +1,227 @@
+func (h *History) ConfigureSigningKey(signingKey *ecdsa.PrivateKey) {
+	h.signingKey = signingKey
+}
+
+// GetManifest returns the manifest for the given hash.
+func (h *History) GetManifest(hash [hashSize]byte) ([]byte, error) {
+	return h.getContentaddressed(hash, "manifests/%s")
+}
+
+// SetManifest sets the manifest and returns its hash.
+func (h *History) SetManifest(manifest []byte) ([hashSize]byte, error) {
+	return h.setContentaddressed("manifests/%s", manifest)
+}
+
+// GetPolicy returns the policy for the given hash.
+func (h *History) GetPolicy(hash [hashSize]byte) ([]byte, error) {
+	return h.getContentaddressed(hash, "policies/%s")
+}
+
+// SetPolicy sets the policy and returns its hash.
+func (h *History) SetPolicy(policy []byte) ([hashSize]byte, error) {
+	return h.setContentaddressed("policies/%s", policy)
+}
+
+// GetTransition returns the transition for the given hash.
+func (h *History) GetTransition(hash [hashSize]byte) (Transition, error) {
+	transitionBytes, err := h.getContentaddressed(hash, "transitions/%s")
+	if err != nil {
+		return Transition{}, err
+	}
+	var transition Transition
+	if err := transition.unmarshalBinary(transitionBytes); err != nil {
+		return Transition{}, fmt.Errorf("unmarshaling transition: %w", err)
+	}
+	return transition, nil
+}
+
+// SetTransition sets the transition and returns its hash.
+func (h *History) SetTransition(transition Transition) ([hashSize]byte, error) {
+	return h.setContentaddressed("transitions/%s", transition.marshalBinary())
+}
+
+// GetLatest returns the verified transition for the given hash.
+func (h *History) GetLatest() (LatestTransition, error) {
+	if h.signingKey == nil {
+		return LatestTransition{}, errors.New("signing key not configured")
+	}
+	transitionBytes, err := h.store.Get("transitions/latest")
+	if err != nil {
+		return LatestTransition{}, fmt.Errorf("getting latest transition: %w", err)
+	}
+	var latestTransition LatestTransition
+	if err := latestTransition.unmarshalBinary(transitionBytes); err != nil {
+		return LatestTransition{}, fmt.Errorf("unmarshaling latest transition: %w", err)
+	}
+	if err := latestTransition.verify(&h.signingKey.PublicKey); err != nil {
+		return LatestTransition{}, fmt.Errorf("verifying latest transition: %w", err)
+	}
+	return latestTransition, nil
+}
+
+// SetLatest signs and sets the latest transition if the current latest is equal to oldT.
+func (h *History) SetLatest(oldT, newT LatestTransition) error {
+	if h.signingKey == nil {
+		return errors.New("signing key not configured")
+	}
+	if err := newT.sign(h.signingKey); err != nil {
+		return fmt.Errorf("signing latest transition: %w", err)
+	}
+	if err := h.store.CompareAndSwap("transitions/latest", oldT.marshalBinary(), newT.marshalBinary()); err != nil {
+		return fmt.Errorf("setting latest transition: %w", err)
+	}
+	return nil
+}
+
+func (h *History) getContentaddressed(hash [hashSize]byte, pathFmt string) ([]byte, error) {
```

Comment:
I think it is common practice for functions that take a format specifier, to have it in the first place.
Or did you tried to make the difference between the set and get version clearer?